### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by 27ManishaPatel**

### CODE REVIEW

File: nodejs/week2/hw-warmup/app.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+//GET /search
+ app.get('/search',(req,res) => {
+    const q = req.query.q;
+    if(q === undefined){
+        res.send(jsonData)
+    }else{
+        jsonData.forEach(obj => {
+            Object.keys(obj).forEach(objKey => {
+                console.log(new String(obj[objKey]))
```
Comment (itsgauravsaxena) [inline nodejs/week2/hw-warmup/app.js]: Always good idea to remove console logs :)

File: nodejs/week2/hw-warmup/app.js [hunk 2]
```diff
@@ -46,11 +46,30 @@ app.get("/documents/:id", (req, res) => {
...
+        Object.keys(fields).forEach((key) => {
+          dataSorted = jsonData.filter((x) => x[key] === fields[key]);
+          res.send(dataSorted);
+        });
+       
+      } else if (q != undefined) {
+        jsonData.forEach((obj) => {
+          Object.keys(obj).forEach((objKey) => {
+            if (new String(obj[objKey]).includes(q))
+            res.send(data);
```
Comment (itsgauravsaxena) [inline nodejs/week2/hw-warmup/app.js]: This `data` will not work. You should return `obj ` instead

**Review by itsgauravsaxena — CHANGES_REQUESTED** [review]

File: nodejs/week2/class-exercise/app.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const app = express();
+const router = express.Router();
```
Comment (andrii1) [inline nodejs/week2/class-exercise/app.js:3]: is it convenient for you to have class exercises in the same branch? I also tried that, but then messed up node modules 😅

File: nodejs/week2/hw-warmup/app.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+//POST /search
+
+app.post("/search", (req, res) => {
+     const q = req.query.q;
+     const fields = req.body.fields
+
+     if (fields !== undefined && q !== undefined) {
+        res.status(400).send("Both query parameters and fields parameters can't be provided");
+
+      }else if (fields != undefined) {
```
Comment (andrii1) [inline nodejs/week2/hw-warmup/app.js:56]: some formatting issues, check if your prettier work ok :)

File: nodejs/week2/class-exercise/app.js (hunk 3 above)
Comment (27ManishaPatel) [reply nodejs/week2/class-exercise/app.js:3]: Yes it is good to record the work and exercise and I messed up with node modules in the beginning but then I learned That how to come out with all that 😁

**Pull request merged**